src/Components/TicTacToe/TicTacToe.js

```diff
@@ -0,0 +1,71 @@
+import React, {useState} from "react"
+import Board from "./Board";
+
+
+export function checkWin(squares, size) {
+
+  // transform the array of squares into a 2D array
+  const matrix = [];
+  for (let i = 0; i < size; i++) {
+    matrix.push(squares.slice(i * size, i * size + size));
+  }
+
+  // check rows for win
+  for (let i = 0; i < size; i++) {
+    const row = matrix[i];
+    if (row.every(e => e === row[0])) {
+      return row[0];
+    }
+  }
+
+  // check columns for win
+  for (let i = 0; i < size; i++) {
+    const column = [];
+    for (let j = 0; j < size; j++) {
+      column.push(matrix[j][i]);
+    }
+    if (column.every(e => e === column[0])) {
+      return column[0];
+    }
+  }
+
+  // check diagonals for win
+  const first_diagonal = [];
+  const second_diagonal = [];
+  for (let i = 0; i < size; i++) {
+    first_diagonal.push(matrix[i][i]);
+    second_diagonal.push(matrix[size - i - 1][size - i - 1]);
+  }
+  if (first_diagonal.every(e => e === first_diagonal[0])) {
+    return first_diagonal[0];
+  }
+  if (second_diagonal.every(e => e === second_diagonal[0])) {
+    return second_diagonal[0];
+  }
+
+  return null;
+}
+
+
+export default function TicTacToe({size = 4}) {
+  const [board, setBoard] = useState(Array(size * size).fill(null));
+  const [currPlayer, setCurrPlayer] = useState(true);
+  const winner = checkWin(board, size);
```

Comment:
Winner should probably be checked in the `handleSetSquare` function, or perhaps in a `useEffect` or `useMemo` hook. Currently, it's run every render, which is fine as you reset the board, but we may not immediately want to do that in the future.

Reply:
When checking inside the `handleSetSquare`, you *could* even optimise the logic to only check for wins including the newly set square (I'll leave it to you to decide whether you want to spend time on that)